libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1158,7 +1158,14 @@ void SMTEncoder::visitStructConstructorCall(FunctionCall const& _funCall)
 	if (smt::isNonRecursiveStruct(*_funCall.annotation().type))
 	{
 		auto& structSymbolicVar = dynamic_cast<smt::SymbolicStructVariable&>(*m_context.expression(_funCall));
-		structSymbolicVar.assignAllMembers(applyMap(_funCall.sortedArguments(), [this](auto const& arg) { return expr(*arg); }));
+		auto structType = dynamic_cast<StructType const*>(structSymbolicVar.type());
+		solAssert(structType, "");
+		auto const& structMembers = structType->structDefinition().members();
+		solAssert(structMembers.size() == _funCall.sortedArguments().size(), "");
+		structSymbolicVar.assignAllMembers(applyMap(
+			_funCall.sortedArguments(),
+			[&, i=0u] (auto const& arg) mutable { return expr(*arg, structMembers[i++]->type()); }
```

Comment:
This would be better with a version of `applyMap` taking two sequences.
Or is there something in `ranges` that could be used here?

Reply:
There seems to be zip in ranges. Here is an example: https://github.com/ethereum/solidity/blob/develop/libsolidity/analysis/TypeChecker.cpp#L983

Reply:
`zip` seems to work and is more readable, I think. Thanks!
I added it as a separate commit for comparison, I can squash if this gets approved.